[Metrics UI] Add docs for alerting context variables (#20)
elastic/observability-docs: docs/en/metrics/metrics-alerting.asciidoc

To provide additional information about an alert, you can include basic and advanced variables in your action message.

[role="screenshot"]
image::images/metrics-alert-message.png[Example alert message]

**Basic variables**

- `alertName`: The name of the alert
- `context.alertState`: The current state of the alert. This value is usually **Alert**. However, if you selected *Alert me if there's no data*, the value can also be **No Data**
- `context.group`: The *group* that the alert message concerns, if you've specified a value in *Create alert per*
- `context.reason`: A verbose description of why the alert is in the reported `alertState`. For example, *my.metric is above a threshold of 1.0 (current value is 1.5)*
- `context.timestamp`: The time at which the message was sent

**Advanced variables**

If you'd like more customization than `context.reason` provides, you can also construct a message with the following advanced variables. 

Using the structure of `context.{advancedVariableName}.condition{n}`, each of these variables is a record containing information about each alert condition. For example, if your alert has two conditions, you can access the value of both using:

> ```
> {{context.value.condition0}}
> {{context.value.condition1}}
> ```

Reviewer: My only suggestion is to use the native code block for asciidcotor. I've chosen text as the content type in the example below, but please change it if that's incorrect.

```suggestion
[source,text]
----
{{context.value.condition0}}
{{context.value.condition1}}
----
```

- Non-aggregatable fields: *matches*, *does not match*, *matches phrase*, *does not match phrase*. 

[role="screenshot"]
image::images/create-metrics-alert.png[Create metrics alert]

[float]
==== Action messages

> You can insert the following variables into your action messages to provide information about an alert:

Reviewer 2: ```suggestion
To provide additional information about an alert, you can include basic and advanced variables in your action message.
```

[role="screenshot"]
image::images/metrics-alert-message.png[Example alert message]

**Basic variables**

> - **alertName**: The name of the alert

Reviewer 2: Instead of highlighting each variable name in **bold**, I suggest using the code syntax. For example, `alertName`.

> - **context.alertState**: The current state of the alert; this will likely be **Alert**, but may also be **No Data** if you've checked off *Alert me if there's no data*

Reviewer 2: ```suggestion
- `context.alertState`: The current state of the alert. This value is usually **Alert**. However, if you selected *Alert me if there's no data*, the value can also be **No Data**
```

- **context.group**: The *group* that the alert message concerns, if you've specified a value in *Create alert per*

> - **context.reason**: A verbose description of why the alert is in the reported **alertState**. This will be something like *my.metric is above a threshold of 1.0 (current value is 1.5)*

Reviewer 2: ```suggestion
- **context.reason**: A verbose description of why the alert is in the reported **alertState**. For example, *my.metric is above a threshold of 1.0 (current value is 1.5)*
```

- **context.timestamp**: The time at which the message was sent

**Advanced variables**

If you'd like more customization than **context.reason** provides, you can also construct a message with the following advanced variables. 

> Each of these variables is a *record* containing information about each *alert condition*, with the structure of **context.{advancedVariableName}.condition{n}**. For example, if your alert has two conditions, you can access the value of both of them using:

Reviewer 2: ```suggestion
Using the structure of **context.{advancedVariableName}.condition{n}**, each of these variables is a record containing information about each alert condition. For example, if your alert has two conditions, you can access the value of both using:
```
